Merge some cnf

richsalz opened the pull request:
This is based on #11338.  This merges a few configuration files into one, and uses flags to identify the right section (hence the dependence on 11338).  It also removes a whole bunch of outdated comments.  I am going through all the test config (er, .cnf :) files in order to make it easier to run the testsuite with just one provider, for example.  This PR is a pre-requisite to that testing.

richsalz commented:
Rebased to fix a conflict in test/recipes/80-test_ca.t 

This also cleans up comments and syntax in several config files, leveraging the doc updates that have already gone in.

richsalz commented:
I am currently having a problem.  Any clues?
```
Request is in newreq.pem, private key is in newkey.pem
../../util/wrap.pl /usr/bin/perl ../../apps/CA.pl -newreq -extra-req '-section userreq' => 0
ok 2 - creating certificate request
====
../../util/wrap.pl ../../apps/openssl ca -rand_serial -inform DER -config "../../apps/openssl.cnf" -policy policy_anything -out newcert.pem -infiles newreq.pem 
Using configuration from ../../apps/openssl.cnf
Unable to load certificate request
80:07:44:AE:7B:7F:00:00:error:asn1 encoding routines:asn1_check_tlen:wrong tag:crypto/asn1/tasn_dec.c:1135:
80:07:44:AE:7B:7F:00:00:error:asn1 encoding routines:asn1_item_embed_d2i:nested asn1 error:crypto/asn1/tasn_dec.c:293:Type=X509_REQ
==> 256
====
ok 3 - signing certificate request
====
../../util/wrap.pl ../../apps/openssl verify -CAfile ./demoCA/cacert.pem newcert.pem 
Can't open newcert.pem for reading, No such file or directory
80:47:0D:11:B5:7F:00:00:error:system library:BIO_new_file:No such file or directory:crypto/bio/bss_file.c:69:calling fopen(newcert.pem, r)
80:47:0D:11:B5:7F:00:00:error:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:77:
Unable to load certificate file
==> 512
```

richsalz commented:
I figured it out.  I had dropped an "-outform DER" when I modified openssl command args.  In doing so, I noticed that quotes around $cnf were missing in the SM2 case, so I fixed two variable assignments to do the right thing.

Please review this.

t8m commented on doc/man7/proxy-certificates.pod:
```diff
@@ -116,7 +116,7 @@ two commands:
     openssl x509 -req -CAcreateserial -in proxy.req -out proxy.crt \
         -CA user.crt -CAkey user.key -days 7 \
-        -extfile proxy.cnf -extensions v3_proxy1
+        -extfile proxy.cnf -extensions proxy
```
Please note that above in the config file snippet example v3_proxy is used as the name of the extension section. So perhaps I would drop this change here, or it should be changed above as well.

richsalz replied:
Fixed the CN's to match; thanks!

t8m replied:
You've misunderstood me - I was talking about the configuration file snippet in the "Creating proxy certificates" section in this manual page.

richsalz replied:
Fixed the config file example too, now.

t8m commented on doc/man7/proxy-certificates.pod:
```diff
@@ -128,7 +128,7 @@ section for the proxy extensions):
     openssl x509 -req -CAcreateserial -in proxy2.req -out proxy2.crt \
         -CA proxy.crt -CAkey proxy.key -days 7 \
-        -extfile proxy.cnf -extensions v3_proxy2
+        -extfile proxy.cnf -extensions proxy_2
```
Similarly here.

t8m commented on test/recipes/80-test_ca.t:
```diff
@@ -18,26 +18,29 @@ use OpenSSL::Test::Utils;
      skip "failed creating new certificate request", 3
 	 if !ok(run(perlapp(["CA.pl","-newreq",
-                             "-extra-req","-outform DER"])),
+                             '-extra-req', '-outform DER -section userreq'])),
```
Just a nit, but why not add the -section userreq as separate list item similarly to the rest.

richsalz replied:
i split it into two extra-req options; hope that's more clear.

t8m replied:
Ah now I get it.

t8m commented on test/recipes/90-test_store.t:
```diff
@@ -16,6 +16,7 @@ my $test_name = "test_store";
 setup($test_name);
 
 my $mingw = config('target') =~ m|^mingw|;
+my $cnf=srctop_file("test","ca-and-certs.cnf");
```
Nit - space after comma?

t8m commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -48,33 +48,27 @@ my @verifycmd = ("openssl", "verify");
 my $Ecert="certE.ss";
 
-my $P1conf=srctop_file("test","P1ss.cnf");
+my $proxycnf=srctop_file("test","proxy.cnf");
```
Nit: space after comma?

t8m commented on test/recipes/80-test_ssl_old.t:
```diff
@@ -48,33 +48,27 @@ my @verifycmd = ("openssl", "verify");
 my @genpkeycmd = ("openssl", "genpkey");
 my $dummycnf = srctop_file("apps", "openssl.cnf");
 
+my $cnf=srctop_file("test","ca-and-certs.cnf");
```
Nit: space after comma?

t8m commented on test/recipes/25-test_verify_store.t:
```diff
@@ -18,34 +18,31 @@ plan tests => 10;
 
 my $dummycnf = srctop_file("apps", "openssl.cnf");
 
+my $cnf=srctop_file("test","ca-and-certs.cnf");
```
Nit: space after comma? And perhaps around =

richsalz commented:
Fixed all the space-after-comma, etc., issues.  Just resolved the comments, to avoid email deluge.  Fixed other things as well.  Fixup commit pushed.  Thanks for the careful read!

richsalz commented:
Rebased and squashed a fixup commit. @t8m needs a re-review, and @openssl-otc needs a review.

t8m commented:
@openssl/committers Ping for the second review.

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -99,32 +97,32 @@ colons between each byte (every second hex digit):
 
-I<NOTE: The proxy policy value is what determines the rights granted
-to the process during the proxy certificate.  It's up to the
+Not that the proxy policy value is what determines the rights granted
```
Not that  ->  Note that

t8m commented:
Still approved after the typo fix.

openssl-machine commented:
This pull request is ready to merge

t8m commented:
Merged to master. Thank you, Rich, for the PR and patience with the reviews.

richsalz commented:
Thank *you*, @t8m for your diligence. You're often the only one and I appreciate it, as do others I am sure.

DDvO commented:
@richsalz, looks like this can be merged.

DDvO commented:
> @richsalz, looks like this can be merged.

Oops, my Firefox tab was partly outdated. So it is already merged :)

levitte commented:
@DDvO, always reload first

DDvO commented:
> @DDvO, always reload first

That's understood, but often I need to reload twice to get the current picture.
Which may be an issue not with Firefox itself but with one of its add-ons/extensions
somehow interfering with the non-trivial web technology used by the GitHub pages.
